### PR TITLE
Make si request in parallel

### PR DESCRIPTION
**Pull request by DeltaEvo**
### Query: "F" (14 pages)
```
Before
si: 15241.757ms
After
si: 6218.550ms
```

### Query: "Darling" (4 pages)
```
Before
si: 4438.146ms
After
si: 2333.308ms
```

### CODE REVIEW

File: src/si/search.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const _ = require('lodash')
 const {extractFromHTML} = require('./scrap.js')
 
 const URI = require('./url.json').url
+const timeout = time => new Promise(resolve => setTimeout(resolve, time))
```
Comment (Kylart) [inline src/si/search.js]:
```javascript
const timeout = (time) => new Promise((resolve) => setTimeout(resolve, time))
```

File: src/si/scrap.js [hunk 2]
```diff
@@ -40,7 +40,11 @@ const extractFromHTML = (data) => {
     results.push(toPush)
   })
 
-  return results
+  if (includeMaxPage) {
+    return { results, maxPage: +$('ul.pagination li:nth-last-child(2) a').text() }
+  } else {
+    return results
+  }
```
Comment (Kylart) [inline src/si/scrap.js]:
```javascript
return includeMaxPage
    ? { results, maxPage: +$('ul.pagination li:nth-last-child(2) a').text() }
    : results
```

File: src/si/search.js [hunk 3]
```diff
@@ -65,28 +66,23 @@ const searchAll = (term = '', opts = {}) => {
...
-
-    while (_continue && page <= 15) {
-      // We stop at page === 15 because nyaa.si offers a maximum of 1000 results
-      // which means 14 pages of 75 results with the last one containing only 25.
-      try {
-        results = _.concat(results, tmpData)
+    try {
+      const { results: fResults, maxPage } = await searchPage(term, 1, opts, true)
+      const searchs = []
+      for (let page = 2; page <= maxPage; page++) {
```
Comment (Kylart) [inline src/si/search.js]: I'm a fervent defender of `++var` instead of `var++` 😄

Comment (DeltaEvo) [reply src/si/search.js]: xD

**Pull request merged**